appcues/src/main/java/com/appcues/ui/extensions/DrawScopeEx.kt

```diff
@@ -11,39 +9,25 @@ import java.lang.Float.max
 /**
  * Defines a path to draw the X by drawing two lines crossing from edge to edge.
  */
-internal fun DrawScope.xShapePath(
-    color: Color,
-    pathSize: Dp,
-    strokeWidth: Dp,
-): Path {
+internal fun DrawScope.xShapePath(pathSize: Dp): Path {
     return Path()
         .apply {
-            val strokeWidthPx = strokeWidth.toPx()
             val pathSizePx = pathSize.toPx()
+            val sizeRect = size.toRect()
 
             // center the desired size within the current bounds
+            // find the delta to deflate the container rect in each direction
             val deltaX = max(size.width - pathSizePx, 0.0f) / 2
             val deltaY = max(size.height - pathSizePx, 0.0f) / 2
 
-            val sizeRect = size.toRect()
-            val pathRect = Rect(
-                left = sizeRect.left + deltaX,
-                top = sizeRect.top + deltaY,
-                right = sizeRect.right - deltaX,
-                bottom = sizeRect.bottom - deltaY
-            )
+            val minX = 0 + deltaX
+            val minY = 0 + deltaY
+            val maxX = sizeRect.right - deltaX
+            val maxY = sizeRect.bottom - deltaY
 
-            drawLine(
-                color = color,
-                start = pathRect.topLeft,
-                end = pathRect.bottomRight,
-                strokeWidth = strokeWidthPx,
-            )
-            drawLine(
-                color = color,
-                start = pathRect.bottomLeft,
-                end = pathRect.topRight,
-                strokeWidth = strokeWidthPx,
-            )
```

Comment:
the issue with what was happening before here - this was actually drawing the Path, not just creating it! These `DrawScope.drawLine(...)` calls paint the path. I updated this to simply create and return the Path, and then let the callsite draw the path. That way, I could have fine control over what color and draw style get used, based on the styling options.

Reply:
interesting fact, i had color being passed down twice basically because of this, I dont remember exactly how to tried to draw it using the lineTo but I had no success for some reason. Nice work here